From cf20644ee522f0d7917d5debd1cad4d1b1cea314 Mon Sep 17 00:00:00 2001
From: Lukas Mrtvy <lukas.mrtvy@gmail.com>
Date: Wed, 18 Jul 2018 23:07:41 +0200
Subject: [PATCH] procd: change noatime to relatime for unprivileged lxc

Signed-off-by: Lukas Mrtvy <lukas.mrtvy@gmail.com>
---
 package/system/procd/initd/early.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/package/system/procd/initd/early.c b/package/system/procd/initd/early.c
index 00fd946..ca66b98 100644
--- a/package/system/procd/initd/early.c
+++ b/package/system/procd/initd/early.c
@@ -56,22 +56,22 @@ early_mounts(void)
 {
 unsigned int oldumask = umask(0);
 
-mount("proc", "/proc", "proc", MS_NOATIME | MS_NODEV | MS_NOEXEC | MS_NOSUID, 0);
-mount("sysfs", "/sys", "sysfs", MS_NOATIME | MS_NODEV | MS_NOEXEC | MS_NOSUID, 0);
+mount("proc", "/proc", "proc", MS_RELATIME | MS_NODEV | MS_NOEXEC | MS_NOSUID, 0);
+mount("sysfs", "/sys", "sysfs", MS_RELATIME | MS_NODEV | MS_NOEXEC | MS_NOSUID, 0);
 mount("cgroup", "/sys/fs/cgroup", "cgroup",  MS_NODEV | MS_NOEXEC | MS_NOSUID, 0);
-mount("tmpfs", "/dev", "tmpfs", MS_NOATIME | MS_NOSUID, "mode=0755,size=512K");
+mount("tmpfs", "/dev", "tmpfs", MS_RELATIME | MS_NOSUID, "mode=0755,size=512K");
 ignore(symlink("/tmp/shm", "/dev/shm"));
 mkdir("/dev/pts", 0755);
-mount("devpts", "/dev/pts", "devpts", MS_NOATIME | MS_NOEXEC | MS_NOSUID, "mode=600");
+mount("devpts", "/dev/pts", "devpts", MS_RELATIME | MS_NOEXEC | MS_NOSUID, "mode=600");
 early_dev();
 
 early_console("/dev/console");
 if (mount_zram_on_tmp()) {
-mount("tmpfs", "/tmp", "tmpfs", MS_NOSUID | MS_NODEV | MS_NOATIME, 0);
+mount("tmpfs", "/tmp", "tmpfs", MS_NOSUID | MS_NODEV | MS_RELATIME, 0);
 mkdir("/tmp/shm", 01777);
 } else {
 mkdir("/tmp/shm", 01777);
-mount("tmpfs", "/tmp/shm", "tmpfs", MS_NOSUID | MS_NODEV | MS_NOATIME,
+mount("tmpfs", "/tmp/shm", "tmpfs", MS_NOSUID | MS_NODEV | MS_RELATIME,
 "mode=01777");
 }
 mkdir("/tmp/run", 0755);
-- 
2.17.1
