From 921afe54a31cc3bc309a3dd03832f1adc70eba25 Mon Sep 17 00:00:00 2001
Date: Mon, 16 Jul 2018 15:35:14 +0200
Subject: [PATCH] package: lxc bump to 3.0.1

---
 feeds/packages/utils/lxc/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/feeds/packages/utils/lxc/Makefile b/feeds/packages/utils/lxc/Makefile
index 288d5039..59b086c9 100644
--- a/feeds/packages/utils/lxc/Makefile
+++ b/feeds/packages/utils/lxc/Makefile
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=lxc
-PKG_VERSION:=2.1.1
+PKG_VERSION:=3.0.1
 PKG_RELEASE:=1
 
 PKG_LICENSE:=LGPL-2.1+ BSD-2-Clause GPL-2.0
@@ -16,7 +16,7 @@ PKG_MAINTAINER:=Marko Ratkaj <marko.ratkaj@sartura.hr>
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://linuxcontainers.org/downloads/lxc/
-PKG_HASH:=68663a67450a8d6734e137eac54cc7077209fb15c456eec401a2c26e6386eff6
+PKG_HASH:=45986c49be1c048fa127bd3e7ea1bd3347e25765c008a09a2e4c233151a2d5db
 
 PKG_BUILD_DEPENDS:=lua
 PKG_BUILD_PARALLEL:=1
-- 
2.14.1
