--- a/leds-apu2.c
+++ b/leds-apu2.c
@@ -328,7 +328,12 @@ static int __init gpio_apu2_init (void)
        const char *board_name = dmi_get_system_info(DMI_BOARD_NAME);
 
        /* Match the device name/model */
-       if (!board_name || !board_vendor || strcasecmp(board_vendor, "PC Engines") || strcasecmp(board_name, "apu2")) {
+       if (!board_name || !board_vendor ||
+               strcasecmp(board_vendor, "PC Engines") ||
+               (strcasecmp(board_name, "PCEngines apu2") &&
+               strcasecmp(board_name, "PC Engines apu2") &&
+               strcasecmp(board_name, "APU2"))
+       ) {
                err = -ENODEV;
                goto exit;
        }
